src/Models/Classes/AdapterDataClasses/AzureManagementData.ts

```diff
@@ -30,9 +30,9 @@ export class AzureResourcesData implements IAdapterData {
 }
 
 export class AzureMissingRoleDefinitionsData implements IAdapterData {
-    data: MissingAzureRoleDefinitionAssignments;
+    data: AzureAccessPermissionRoleGroups;
```

Comment:
curious why this change was needed. It seems like it's larger than just fixing the type mismatch now that we have the grouped array.

Reply:
I renamed it to refer to a "group" including "enforced" and "interchangeables" instead - not to frame it just for "missing" roles, now it is referring just a group of roles I am interested in basically, it can be missing roles or it can refer to a group that I want to check against. Also, I moved the missing role checking related methods to the Utils file to make it consolidated (single place to be consumed by many components/adapters) and kept the internal logic same that I used as a fix last week